cmd/operator-sdk/bundle/add.go

```diff
@@ -0,0 +1,227 @@
+				binaryPath += ".exe"
+			}
+			if err := c.buildOPM(binaryPath); err != nil {
+				log.Fatalf("Error building image builder: %v", err)
+			}
+			if err := c.runOPMIndexAdd(binaryPath); err != nil {
+				log.Fatalf("Error buildingindex image: %v", err)
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&c.fromIndex, "from-index", "f", "", "Previous index to build new index image from")
+	cmd.Flags().StringVarP(&c.toIndex, "to-index", "t", "", "Tag for new index image being built")
+	if err := cmd.MarkFlagRequired("to-index"); err != nil {
+		panic(err)
+	}
+	cmd.Flags().StringVar(&c.dockerfileName, "dockerfile-name", "",
+		"Name of the Dockerfile to generate if --generate-only is set. Default is 'Dockerfile'")
+	cmd.Flags().StringVar(&c.imageBuilder, "image-builder", "docker",
+		"Tool to build container images. One of: [docker, podman]")
+	cmd.Flags().BoolVarP(&c.generateOnly, "generate-only", "g", false,
+		"Generate metadata and a Dockerfile on disk without building the bundle image")
+	cmd.Flags().BoolVar(&c.permissive, "permissive", false,
+		"Allow registry load errors without exiting the build")
+
+	return cmd
+}
+
+func (c bundleIndexCmd) validate() error {
+	if !c.generateOnly && c.dockerfileName != "" {
+		return fmt.Errorf("dockerfile name can only be set if generating files")
+	}
+	return nil
+}
+
+// buildOPM reates and download 'opm' to a binary directory using 'go get' to
+// download a particular version, specified by registryVersion, to install.
+// TODO(estroz): shell out to make after kubebuilder integration.
+func (c bundleIndexCmd) buildOPM(binaryPath string) error {
+	binaryDir := filepath.Dir(binaryPath)
+	if err := os.MkdirAll(binaryDir, fileutil.DefaultDirFileMode); err != nil {
+		return err
+	}
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	if err := os.Chdir(binaryDir); err != nil {
+		return err
+	}
+	defer func() {
+		if err := os.Chdir(wd); err != nil {
+			log.Fatal(err)
+		}
+	}()
+	defer func() {
+		if err := os.RemoveAll("go.mod"); err != nil {
+			log.Fatal(err)
+		}
+	}()
+	defer func() {
+		if err := os.RemoveAll("go.sum"); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	cmds := []*exec.Cmd{
+		exec.Command("go", "mod", "init"),
+		exec.Command("go", "get", fmt.Sprintf("%s@%s", opmCmdPath, registryVersion)),
+		exec.Command("go", "build", "-o", filepath.Base(binaryPath), opmCmdPath),
+	}
+	for _, cmd := range cmds {
+		if err := projutil.ExecCmd(cmd); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// runOPMIndexAdd runs 'opm index add', setting flags from values in c. No
+// files are generated unless generateOnly is true.
+// Note: operator-registry sports a Go library to handle index building, but
+// it requires operator-sdk be built with CGO_ENABLED=1. This is currently
+// not possible due to k8s dependency requirements.
+func (c bundleIndexCmd) runOPMIndexAdd(binaryPath string) error {
+	// Always remove the database, an artifact users do not need to know about.
```

Comment:
Doesn't this conflict with the concept of the `--generate-only` flag? The database is needed to actually use the dockerfile.